pennylane/measurements/probs.py

```diff
@@ -184,93 +184,7 @@ def process_samples(
         return qml.math.squeeze(prob) if bin_size is None else prob
 
     def process_state(self, state: Sequence[complex], wire_order: Wires):
-        num_wires = len(wire_order)
-        batch_size = self._compute_batch_size(state, num_wires)
-        shape = (2**num_wires,) if batch_size is None else (batch_size, 2**num_wires)
-        flat_state = qml.math.reshape(state, shape)
-        real_state = qml.math.real(flat_state)
-        imag_state = qml.math.imag(flat_state)
-        return self.marginal_prob(real_state**2 + imag_state**2, wire_order)
-
-    @staticmethod
-    def _compute_batch_size(state, num_wires):
-        """This logic is isolated because tf.function cannot compile without the try-except
-        clause below (it raises a tf.errors.OperatorNotAllowedInGraphError because it can't
-        handle a function that sometimes returns a tuple, and other times returns None)."""
-
-        expected_shape = [2] * num_wires
-        expected_size = 2**num_wires
-        size = qml.math.size(state)
-
-        try:
-            if qml.math.ndim(state) > len(expected_shape) or size > expected_size:
-                return size // expected_size
-        except Exception as err:  # pylint:disable=broad-except
-            if not qml.math.is_abstract(state):
-                raise err
-
-        return None
-
-    @staticmethod
-    def _count_samples(indices, batch_size, dim):
-        """Count the occurrences of sampled indices and convert them to relative
-        counts in order to estimate their occurrence probability."""
-        num_bins, bin_size = indices.shape[-2:]
-        if batch_size is None:
-            prob = qml.math.zeros((dim, num_bins), dtype="float64")
-            # count the basis state occurrences, and construct the probability vector for each bin
-            for b, idx in enumerate(indices):
-                basis_states, counts = qml.math.unique(idx, return_counts=True)
-                prob[basis_states, b] = counts / bin_size
-
-            return prob
-
-        prob = qml.math.zeros((batch_size, dim, num_bins), dtype="float64")
-        indices = indices.reshape((batch_size, num_bins, bin_size))
-
-        # count the basis state occurrences, and construct the probability vector
-        # for each bin and broadcasting index
-        for i, _indices in enumerate(indices):  # First iterate over broadcasting dimension
-            for b, idx in enumerate(_indices):  # Then iterate over bins dimension
-                basis_states, counts = qml.math.unique(idx, return_counts=True)
-                prob[i, basis_states, b] = counts / bin_size
-
-        return prob
-
-    def marginal_prob(self, prob, wire_order):
```

Comment:
It really only exists as an artifact of copy-pasting this code from default.qubit. If I left it, this file would look like:
```python
def process_state(self, state, wires, ...):
    return self.marginal_prob(qml.math.real(state)**2 + qml.math.imag(state)**2, wires)

def marginal_prob(prob, wires):
    ...
```

so like, what's the point of having these are different functions? plus - no one uses `marginal_prob` because although it's public, it's not part of the `StateMeasurement` API so no one even knows it exists. If you're really worried, I can add it back, but imo it's unnecessary

Reply:
Nope, that makes sense to me. I'd run it by some other people as well though, and also add a changelog entry. I'd rather not assume that *no one* was using `ProbabilityMP.marginal_probs`.

Reply:
added an entry to the breaking changes section